dvc/external_repo.py

```diff
@@ -127,9 +127,17 @@ def _fix_upstream(self):
         if not os.path.isdir(self.url):
             return
```

Comment:
This check is redundant here as it should be covered by below `try/except` block. But, I think, this makes the intention clearer, so, keeping it for now. :)